satpy/readers/sar_c_safe.py

```diff
@@ -704,3 +689,110 @@ def start_time(self):
     def end_time(self):
         """Get the end time."""
         return self._end_time
+
+
+class SAFESARReader(GenericYAMLReader):
+    """A reader for SAFE SAR-C data for Sentinel 1 satellites."""
+
+    def __init__(self, config, filter_parameters=None):
+        """Set up the SAR reader."""
+        super().__init__(config)
+        self.filter_parameters = filter_parameters
+        self.files_by_type = defaultdict(list)
+        self.storage_items = []
+
+    @property
+    def start_time(self):
+        """Get the start time."""
+        return self.storage_items.values()[0].filename_info["start_time"].replace(tzinfo=tz.utc)
+
+    @property
+    def end_time(self):
+        """Get the end time."""
+        return self.storage_items.values()[0].filename_info["end_time"].replace(tzinfo=tz.utc)
+
+    def load(self, dataset_keys, **kwargs):
+        """Load some data."""
+        if kwargs:
+            raise NotImplementedError(f"Don't know how to handle kwargs {kwargs}")
```

Comment:
Doesn't this prevent using another reader at the same time if that reader was provided kwargs at load time? Or maybe that just means a user has to provide them per-reader...if that's possible.

Reply:
I thought `reader_kwargs` was of the form `{"abi_l1b": ..., "ahi_hsd": ...}, and use that assumption when writing this. I'll check

Reply:
from the `scene.__init__` docstring:

> Keyword arguments to pass to specific reader instances.
                Either a single dictionary that will be passed onto to all
                reader instances, or a dictionary mapping reader names to
                sub-dictionaries to pass different arguments to different
                reader instances.

So we are both right... I'd like to have it just as sub dictionaries, would be cleaner, but I'll fix the code here for now.